registration.php

```diff
@@ -1,407 +1,874 @@
-                                    <option value="Chandigarh">Chandigarh</option>
-                                    <option value="Chhattisgarh">Chhattisgarh</option>
-                                    <option value="Dadra and Nagar Haveli">Dadra and Nagar Haveli</option>
-                                    <option value="Daman and Diu">Daman and Diu</option>
-                                    <option value="Delhi">Delhi</option>
-                                    <option value="Goa">Goa</option>
-                                    <option value="Gujarat">Gujarat</option>
-                                    <option value="Haryana">Haryana</option>
-                                    <option value="Himachal Pradesh">Himachal Pradesh</option>
-                                    <option value="Jammu and Kashmir">Jammu and Kashmir</option>
-                                    <option value="Jharkhand">Jharkhand</option>
-                                    <option value="Karnataka">Karnataka</option>
-                                    <option value="Kerala">Kerala</option>
-                                    <option value="Lakshadweep">Lakshadweep</option>
-                                    <option value="Madhya Pradesh">Madhya Pradesh</option>
-                                    <option value="Maharashtra">Maharashtra</option>
-                                    <option value="Manipur">Manipur</option>
-                                    <option value="Meghalaya">Meghalaya</option>
-                                    <option value="Mizoram">Mizoram</option>
-                                    <option value="Nagaland">Nagaland</option>
-                                    <option value="Orissa">Orissa</option>
-                                    <option value="Pondicherry">Pondicherry</option>
-                                    <option value="Punjab">Punjab</option>
-                                    <option value="Rajasthan">Rajasthan</option>
-                                    <option value="Sikkim">Sikkim</option>
-                                    <option value="Tamil Nadu">Tamil Nadu</option>
-                                    <option value="Telangana">Telangana</option>
-                                    <option value="Tripura">Tripura</option>
-                                    <option value="Uttar Pradesh">Uttar Pradesh</option>
-                                    <option value="Uttarakhand">Uttarakhand</option>
-                                    <option value="West Bengal">West Bengal</option>
+                                    <option value="">Select State
+                                    </option>
+                                    <option 
+                                       <?php if(isset($home_state) && $home_state=='Andaman and Nicobar Islands') echo 'selected'; ?> value="Andaman and Nicobar Islands">Andaman and Nicobar Islands
```

Comment:
Instead of writing so many options one by one for each state name, better to store all states as a list in a variable in constant file or may be a statelist.php file and then read the statelist from variable by looping through it.
in this way code size will reduce and will not have to write so many <option> in the code.
also in the loop, create a variable to store home_state and match with the statelist to see if the option needs to be selected or not by using that variable.

something like
<?php
foreach($state_list as $state_name)
{
    $selected = (isset($home_state) && $home_state=== $state_name) ? 'selected' : '';
?>
    <option value='<?php echo $state_name; ?>' <?php echo $selected; ?>>'<?php echo $state_name; ?></option>
<?php
}
?>